### PR TITLE
feat: handle large zipfiles

### LINKED ISSUE #167: Failing build: Failed to upload file: __ssr Failed to upload file: __dsg
**Describe the bug**

Build fails for `gatsby v4.1.0` and `@netlify/plugin-gatsby@2.0.0-zz-beta.0` while works with stable version of @netlify/plugin-gatsby (of course without dsg and ssr)

Here the logs:
```
10:54:59 AM: ────────────────────────────────────────────────────────────────
10:54:59 AM:   6. Deploy site                                                
10:54:59 AM: ────────────────────────────────────────────────────────────────
10:54:59 AM: ​
10:54:59 AM: Starting to deploy site from 'public'
10:55:14 AM: Creating deploy tree asynchronously
10:55:14 AM: Creating deploy upload records
10:55:20 AM: 79 new files to upload
10:55:20 AM: 3 new functions to upload
10:55:26 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:26 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:28 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:28 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:30 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:30 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:31 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:32 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:34 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:35 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:38 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:38 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:42 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:42 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:50 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:51 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:58 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:55:59 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:56:12 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:56:18 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:56:23 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:56:34 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:56:51 AM: Request must be smaller than 69905067 bytes for the CreateFunction operation
10:56:51 AM: Failed to upload file: __ssr
10:56:52 AM: Failed to upload file: __dsg
10:56:52 AM: Site deploy was successfully initiated
10:56:52 AM: Execution cancelled
10:56:52 AM: Error running command: Command was cancelled
10:56:52 AM: Failing build: Failed to build site
```

**Environment**
  Netlify CI:
     node v14.18.1 (npm v6.14.15)
     @netlify/build 18.21.3
  npmPackages:
    @netlify/plugin-gatsby: ^2.0.0-beta => 2.0.0-zz-beta.0 
    gatsby: ^4.1.0 => 4.1.0

### PR DESCRIPTION
**Pull request by ascorbic**
Adds logging of largest files if the function zips are larger than 50MB. Also excludes all `*js.map` files by default.

Fixes #167

### CODE REVIEW

**Review by tiffanosaurus — DISMISSED** [review]

File: plugin/src/helpers/verification.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  }
+  // We don't fail the build, because the actual hard max size is larger so it might still succeed
+  console.log(
+    redBright(stripIndent`
+      The function zip ${yellowBright(
+        relative(process.cwd(), file),
+      )} size is ${prettyBytes(
+      fileSize,
+    )}, which is larger than the maximum supported size of ${prettyBytes(
+      maxSize,
```
[the remark this answers is not in the document]
Comment (ascorbic) [reply plugin/src/helpers/verification.ts:34]: Looks like this is currently undocumented, apart from our note in the Next.js repo. I've raised this with docs, and will add it to the plugin docs in a follow-up

File: plugin/src/helpers/verification.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    )}.
+      There are a few reasons this could happen, such as accidentally bundling a large dependency or adding lots of files to "included_files".
+    `),
+  )
+  const zip = new StreamZip({ file })
+  console.log(`Contains ${await zip.entriesCount} files`)
+  const sortedFiles = Object.values(await zip.entries()).sort(
+    (entryA, entryB) => entryB.size - entryA.size,
```
Comment (tiffanosaurus) [inline plugin/src/helpers/verification.ts:43]: You should close the file according to the docs

```
zip.close()
```

Comment (ascorbic) [reply plugin/src/helpers/verification.ts:43]: That'll need changing in the Next plugin too.

Comment (ascorbic) [reply plugin/src/helpers/verification.ts:43]: OK, fixed

**Review by tiffanosaurus — APPROVED** [review]
zipitandshipit

File: plugin/src/helpers/verification.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    redBright(stripIndent`
+      The function zip ${yellowBright(
+        relative(process.cwd(), file),
+      )} size is ${prettyBytes(
+      fileSize,
+    )}, which is larger than the maximum supported size of ${prettyBytes(
+      maxSize,
+    )}.
+      There are a few reasons this could happen. You may have accidentally bundled a large dependency, or you might have a
+      large number of pre-rendered pages included.
```
[the remark this answers is not in the document]
Comment (KyleBlankRollins) [reply plugin/src/helpers/verification.ts]: Looks like this change is outdated and doesn't match. And the PR is merged, so I assume you don't need a copy review anymore?

**Pull request merged**